stdlib/Serialization/src/Serialization.jl

```diff
@@ -1319,6 +1308,61 @@ function deserialize(s::AbstractSerializer, t::DataType)
         end
         return x
     end
+    if nf == 0
+        x = ccall(:jl_new_struct_uninit, Any, (Any,), t)
+        return x
+    end
+    na = nf
+    #if nf > 4
+        vflds = Vector{Any}(undef, nf)
+        for i in 1:nf
+            tag = Int32(read(s.io, UInt8)::UInt8)
+            if tag != UNDEFREF_TAG
+                f = handle_deserialize(s, tag)
+                na >= i && (vflds[i] = f)
+            else
+                na >= i && (na = i - 1) # rest of tail must be undefined values
+            end
+        end
+        return ccall(:jl_new_structv, Any, (Any, Ptr{Any}, UInt32), t, vflds, na)
+    #else
+    #    ## optimization for small immutable objects--is this worthwhile?
+    #    sflds = SmallArray()
+    #    tag = Int32(read(s.io, UInt8)::UInt8)
+    #    if tag != UNDEFREF_TAG
+    #        sflds.eins = handle_deserialize(s, tag)
+    #    else
+    #        na = 0
+    #    end
+    #    if nf > 1
+    #        tag = Int32(read(s.io, UInt8)::UInt8)
+    #        if tag != UNDEFREF_TAG
+    #            sflds.zwei = handle_deserialize(s, tag)
+    #        else
+    #            na > 1 && (na = 1)
+    #        end
+    #        if nf > 2
+    #            tag = Int32(read(s.io, UInt8)::UInt8)
+    #            if tag != UNDEFREF_TAG
+    #                sflds.drei = handle_deserialize(s, tag)
+    #            else
+    #                na > 2 && (na = 2)
+    #            end
+    #            if nf > 3
+    #                tag = Int32(read(s.io, UInt8)::UInt8)
+    #                if tag != UNDEFREF_TAG
+    #                    sflds.vier = handle_deserialize(s, tag)
+    #                else
+    #                    na > 3 && (na = 3)
+    #                end
+    #            end
+    #        end
+    #    end
+    #    fldptr = Ptr{Any}(pointer_from_objref(sflds))
+    #    GC.@preserve sflds begin
+    #        return ccall(:jl_new_structv, Any, (Any, Ptr{Any}, UInt32), t, fldptr, na)
+    #    end
+    #end
```

Comment:
Please don’t leave large commented out blocks.

Reply:
Agreed; it's better to either include it or leave the code on a branch for later performance testing and PR'ing.